subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy

```diff
@@ -114,6 +114,18 @@ trait ValidationMessageChecker {
             .render()
     }
 
+    @ValidationTestFor(
+        ValidationProblemId.ANNOTATION_INVALID_IN_CONTEXT
+    )
+    String modifierAnnotationInvalidInContext(@DelegatesTo(value = AnnotationContext, strategy = Closure.DELEGATE_FIRST) Closure<?> spec = {}) {
+        def config = display(AnnotationContext, 'annotation_invalid_in_context', spec)
+        config.description("is annotated with invalid modifier @${config.annotation}")
+            .reason("The '@${config.annotation}' annotation cannot be used in this context")
+            .solution("Remove the annotation")
+            .solution("use a different annotation, e.g one of ${config.validAnnotations}")
```

Comment:
My main issue with this kind of testing is that it's really not obvious whether it's a problem that one `solution()` is capitalized, while the other isn't. I don't see the final rendered message anywhere as it is seen by the user.

Reply:
Yeah, I realized you don't have the rendered solution because I forgot to add this test 🤦 

And in the rendered version it seems like all of the solutions are capitalized, so it doesn't matter here.

Reply:
I still think this should be the same capitalization. It costs nothing to fix it, and it's one fewer inconsistency.